src/jsifier.js

```diff
@@ -410,6 +413,10 @@ function ${name}(${args}) {
         contentText = `var ${mangled} = ${snippet};`;
       }
       const sig = LibraryManager.library[symbol + '__sig'];
+      isAsyncFunction = LibraryManager.library[symbol + '__async'] ?? isAsyncFunction;
+      if (isAsyncFunction) {
+        asyncFuncs.push(symbol);
+      }
```

Comment:
Is there a test for this? I searched for `__async` and don't seem to see one.

Reply:
I've moved these changes out into #19093. Once that lands I'll update here.